File: leetcode/two-sum.py
class Solution:
    def twoSum(self, nums, target: int):
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if (nums[i] + nums[j]) == target:
                    return [i, j]

class Solution2:
    def twoSum(self, nums: [int], target: int) -> [int]:
        s = {}
        for i in range(len(nums)):
            t = target - nums[i]
            if t in s and i != s[t]:
                return [i, s[t]]
            s[nums[i]] = i
        return []

print(Solution2().twoSum([2, 5, 5, 11], 10))

File: leetcode/two-sum.java
import java.util.*;


public class Solution {
    /**
     * 
     * @param numbers int整型一维数组 
     * @param target int整型 
     * @return int整型一维数组
     */
    public int[] twoSum (int[] numbers, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < numbers.length; i++) {
            int temp = target - numbers[i];
            if (map.containsKey(numbers[i])) {
                return new int[]{map.get(numbers[i]), i};
            } else {
                map.put(temp, i);
            }
        }
        return new int[2];
    }
}